pkg/trace/statsagent/stats_agent.go

```diff
@@ -0,0 +1,187 @@
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-agent/pkg/trace/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/trace/timing"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type StatsAgent interface {
+	Start()
+	Stop()
+	ComputeStats(ctx context.Context, traces ptrace.Traces)
+}
+
+type StatsAgentConfig struct {
+	ComputeStatsBySpanKind bool
+	PeerTagsAggregation    bool
+	SpanNameAsResourceName bool
+	SpanNameRemappings     map[string]string
+	IgnoreResources        []string
+	PeerTags               []string
+}
+
+type statsAgent struct {
+	out chan *pb.StatsPayload
+	*agent.Agent
+
+	// pchan specifies the channel that will be used to output Datadog Trace Agent API Payloads
+	// resulting from ingested OpenTelemetry spans.
+	pchan chan *api.Payload
+
+	// wg waits for all goroutines to exit.
+	wg sync.WaitGroup
+
+	// exit signals the agent to shut down.
+	exit chan struct{}
+}
+
+func NewAgent(ctx context.Context, cfg *StatsAgentConfig, out chan *pb.StatsPayload, statsd statsd.ClientInterface, translator *attributes.Translator) (StatsAgent, error) {
+	acfg := traceconfig.New()
+	acfg.OTLPReceiver.SpanNameRemappings = cfg.SpanNameRemappings
+	acfg.OTLPReceiver.SpanNameAsResourceName = cfg.SpanNameAsResourceName
+	acfg.Ignore["resource"] = cfg.IgnoreResources
+	acfg.ComputeStatsBySpanKind = cfg.ComputeStatsBySpanKind
+	acfg.PeerTagsAggregation = cfg.PeerTagsAggregation
+	acfg.PeerTags = cfg.PeerTags
+
+	// disable the HTTP receiver
+	acfg.ReceiverPort = 0
+	// set the API key to succeed startup; it is never used nor needed
+	acfg.Endpoints[0].APIKey = "skip_check"
+	// set the default hostname to the translator's placeholder; in the case where no hostname
+	// can be deduced from incoming traces, we don't know the default hostname (because it is set
+	// in the exporter). In order to avoid duplicating the hostname setting in the processor and
+	// exporter, we use a placeholder and fill it in later (in the Datadog Exporter or Agent OTLP
+	// Ingest). This gives a better user experience.
+	acfg.Hostname = metrics.UnsetHostnamePlaceholder
+	acfg.OTLPReceiver.AttributesTranslator = translator
+	pchan := make(chan *api.Payload, 1000)
+	a := agent.NewAgent(ctx, acfg, telemetry.NewNoopCollector(), statsd)
+	// replace the Concentrator (the component which computes and flushes APM Stats from incoming
+	// traces) with our own, which uses the 'out' channel.
+	a.Concentrator = stats.NewConcentrator(acfg, out, time.Now(), statsd)
+	// ...and the same for the ClientStatsAggregator; we don't use it here, but it is also a source
+	// of stats which should be available to us.
+	a.ClientStatsAggregator = stats.NewClientStatsAggregator(acfg, out, statsd)
```

Comment:
Remove this? I don't think we want the client stats aggregator created at all anymore.